pkg/controller/controller.go

```diff
@@ -589,13 +603,13 @@ func (p *csiProvisioner) Provision(options controller.ProvisionOptions) (*v1.Per
 		if err != nil {
 			klog.Warningf("failed to translate CSI PV to in-tree due to: %v. Deleting provisioned PV", err)
 			p.Delete(pv)
-			return nil, err
+			return nil, controller.ProvisioningFinished, err
```

Comment:
We don't check return value of `p.Delete()`. What if that deletion failed partially?

Reply:
Fixed